extensions/ql-vscode/src/archive-filesystem-provider.ts

```diff
@@ -100,6 +100,14 @@ class InvalidSourceArchiveUriError extends Error {
 
 /** Decodes an encoded source archive URI into its corresponding paths. Inverse of `encodeSourceArchiveUri`. */
 export function decodeSourceArchiveUri(uri: vscode.Uri): ZipFileReference {
+  if (!uri.authority) {
```

Comment:
Why are we producing URIs with empty authority in the first place? I would expect them to have authority `0-length` or `1-length`, indicating that the entire URI is the zip path, and the file path within the zip is empty.

Reply:
It is happening from right here:

https://github.com/github/vscode-codeql/pull/629/files#diff-34fd6ff97f21fd8bfa8c93666587f649e2805898a9903e9caf6ee6c528964575R125

Reply:
I see. Could we replace that with a call to `encodeSourceArchiveUri({'sourceArchiveZipPath: zipPath, 'pathWithinSourceArchive': ''})` then? Right now it looks like we're attempting to `decode` a URI that we didn't create with `encode`, which looks like a nice way to make future mistakes :laughing: 

None of this should seriously affect behaviour, but it might be clearer to preserve the pattern that `encode/decode` are inverses.

Reply:
In short, this PR looks reasonable and I'm convinced it fixes the crash, but I think that creating `codeql-zip-archive` URIs manually in the first place without going through the `encode` function is the real bug.

Reply:
I can fix that section to ensure we use encode, but I still think we should be handling uris without an authority. It wasn't easy to track down where this malformed uri came from and it could happen again.

This is a good instance of where we should follow the robustness principle. Be general in what you accept and strict in what you produce. 

And if we see a malformed uri, we can log it.